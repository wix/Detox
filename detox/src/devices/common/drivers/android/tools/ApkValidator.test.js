describe('APK validation', () => {

  const binaryPath = 'mock-bin-path';
  const testBinaryPath = 'mock-test-bin-path';

  let aapt;
  let uut;
  beforeEach(() => {
    const AAPT = jest.genMockFromModule('../exec/AAPT');
    aapt = new AAPT();

    const ApkValidator = require('./ApkValidator');
    uut = new ApkValidator(aapt);
  });

  const givenAAPTResult = (result) => aapt.isTestAPK.mockResolvedValue(result);
  const givenAAPTError = (error) => aapt.isTestAPK.mockRejectedValue(error);

  describe('App APK validation', () => {
    it('should validate the APK is not a test APK', async () => {
      givenAAPTResult(false);
      await uut.validateAppApk(binaryPath);
      expect(aapt.isTestAPK).toHaveBeenCalledWith(binaryPath);
    });

    it('should throw a descriptive error if app APK happens to be a test APK', async () => {
      givenAAPTResult(true);
      await expect(uut.validateAppApk(binaryPath)).rejects.toThrowErrorMatchingSnapshot();
    });

    it('should throw a specific error if AAPT throws', async () => {
      givenAAPTError(new Error('mock error'));
      await expect(uut.validateAppApk(binaryPath)).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe('Test APK validation', () => {
    it('should validate the APK is indeed a test APK', async () => {
      givenAAPTResult(true);
      await uut.validateTestApk(testBinaryPath);
      expect(aapt.isTestAPK).toHaveBeenCalledWith(testBinaryPath);
    });

    it('should throw a descriptive error if APK happens to be an app APK', async () => {
      givenAAPTResult(false);
      await expect(uut.validateTestApk(testBinaryPath)).rejects.toThrowErrorMatchingSnapshot();
    });

    it('should throw a specific error if AAPT throws', async () => {
      givenAAPTError(new Error('mock error'));
      await expect(uut.validateTestApk(testBinaryPath)).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
