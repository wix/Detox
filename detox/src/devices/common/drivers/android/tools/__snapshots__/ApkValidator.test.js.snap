// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APK validation App APK validation should throw a descriptive error if app APK happens to be a test APK 1`] = `
"App APK at path mock-bin-path was detected as the *test* APK!

HINT: Your binary path was probably wrongly set in the active Detox configuration.
For further assistance, visit the Android setup guide: https://github.com/wix/Detox/blob/master/docs/Introduction.Android.md"
`;

exports[`APK validation App APK validation should throw a specific error if AAPT throws 1`] = `
"Failed to get details about the APK at path mock-bin-path. Root cause:
Error: mock error

HINT: Check that the binary path in the active Detox configuration has been set to a path of an APK file.
For further assistance, visit the Android setup guide: https://github.com/wix/Detox/blob/master/docs/Introduction.Android.md"
`;

exports[`APK validation Test APK validation should throw a descriptive error if APK happens to be an app APK 1`] = `
"Test APK at path mock-test-bin-path was detected as the *app* APK!

HINT: Your test test-binary path was probably wrongly set in the active Detox configuration.
For further assistance, visit the Android setup guide: https://github.com/wix/Detox/blob/master/docs/Introduction.Android.md"
`;

exports[`APK validation Test APK validation should throw a specific error if AAPT throws 1`] = `
"Failed to get details about the APK at path mock-test-bin-path. Root cause:
Error: mock error

HINT: Check that the test-binary path in the active Detox configuration has been set to a path of an APK file.
For further assistance, visit the Android setup guide: https://github.com/wix/Detox/blob/master/docs/Introduction.Android.md"
`;
